src/features/speedup/components/SpeedUpMonitor.tsx

```diff
@@ -0,0 +1,56 @@
+import { Alert, AlertTitle, Box, Button, SvgIcon, Typography } from '@mui/material'
+import { SpeedUpModal } from '@/features/speedup/components/SpeedUpModal'
+import Rocket from '@/public/images/common/rocket.svg'
+import { useCounter } from '@/components/common/Notifications/useCounter'
+import { useState } from 'react'
+import { type TransactionDetails } from '@safe-global/safe-gateway-typescript-sdk'
+import { type PendingTx } from '@/store/pendingTxsSlice'
+
+type Props = {
+  txDetails: undefined | TransactionDetails
+  txId: string
+  pendingTx: PendingTx
+  modalTrigger: 'alertBox' | 'alertButton'
+}
+
+const SPEED_UP_THRESHOLD_IN_SECONDS = 15
+
+export function SpeedUpMonitor({ txDetails, txId, pendingTx, modalTrigger = 'alertBox' }: Props) {
+  const [openSpeedUpModal, setOpenSpeedUpModal] = useState(false)
+  const counter = useCounter(pendingTx?.submittedAt)
+
+  if (!counter || counter < SPEED_UP_THRESHOLD_IN_SECONDS) {
```

Comment:
We should also check the pendingTx status to be PROCESSING as we cannot really speed up relaying txs.  
For relayed tx Gelato is handling the gas price and payments.

Reply:
oh, yes. Forgot about this here